### PR TITLE
Closes #1785: Bind `pyzmq<24.0.0`

### LINKED ISSUE #1785: Bind `pyzmq<24.0.0`
Until we can resolve the failures we're seeing with version `24.0.0` in #1784, bind pyzmq to [`23.2.1`](https://pypi.org/project/pyzmq/23.2.1/)

### PR DESCRIPTION
**Pull request by pierce314159**
This PR (Closes #1785):
- Set `pyzmq>=20.0.0,<24.0.0` to avoid failures seen with `24.0.0` in #1784

### CODE REVIEW

**Review by Ethan-DeBandi99 — APPROVED** [review]
Looks good.

**Review by joshmarshall1 — APPROVED** [review]

File: REQUIREMENTS.md [hunk 1]
```diff
@@ -15,7 +15,7 @@ The following python packages are required by the Arkouda client package.
 - `python>=3.8`
 - `numpy>=1.22.2`
 - `pandas>=1.4.0`
-- `pyzmq>=20.0.0`
+- `pyzmq==23.2.1`
```
Comment (ronawho) [inline REQUIREMENTS.md:18]: Do you want this version exactly, or just something >= 20 and < 24? (I'm worried that for instance pypi mirrors we use internally might not have this exact version)

Comment (pierce314159) [reply REQUIREMENTS.md:18]: ooo good point!!

**Review by ronawho — APPROVED** [review]

**Review by Ethan-DeBandi99 — APPROVED** [review]
Looks good

**Pull request merged**